projects/node-http-exercise/node-http-exercise.md

```diff
@@ -0,0 +1,255 @@
+
+2. Create the `/hangup` route, which should close the connection with no response. This one can be created in multiple ways. Think about maybe making a check for this route before the helper function. Remember that you are looking for the `pathname` for this check. See [hints](#hints) for more information about this.
+
+3. Create the `/head` route, which should should respond with 204.
+
+4. Create the `/hello` route, which should should respond with 200 and text body of "Hello".
+
+5. Create the `/echo?message=xyz` route, which should respond with 200 and the message as the body. For this one, to help you out, you should start by console-logging the `req` object to find out how to pull out the message. Also, as a hint, the solution uses a check before the helper function also for that one.
+
+6. Create the `/error` route, which should should respond with 500.
+
+7. Create the `/time.json` route, which should should respond with 200, body with Content-Type of `application/json` and `{ "now": "Sat Feb 15 2020 14:29:20 GMT-0800" }` where it's the current time of the response.
+
+8. Create the `/github` route, which should respond with 302 that redirects to [Techtonica GitHub page](https://github.com/Techtonica).
+
+### Questions to Consider
+
+- Why do you think it is better practice not to hard-code everything within the response object and header?
+- How is this API construct different than a RESTful API?
+- In terms of security, what do you think the implementation with the different checks for routes either in the object or earlier certain endpoints yield? Could this be improved?
+
+### Hints
+
+- You can abruptly end a connection by using `res.end()` and `server.close()`.
+- If called as such, the `Date()` created will look like this: `Wed Oct 14 2020 21:22:50 GMT-0400 (Eastern Daylight Time)`, but for this exercise, we do not want what's in the parentheses at the end. For any more information, I invite you to check the [official Date documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date).
+- If you want to make a check before going into the helper function, you can first create a variable with the route, including the slash, then check if the `pathname` triple-equals to that variable. Then you can either cut the connection shortly or do anything else before your regular routes checks. For example, you could use the `res.end()` and `server.close()` or inquire on the `path.query.message` to get more information on the url.
+
+### Testing
+
+1. If you haven't already done it, [download and install Postman](https://www.postman.com/product/api-client/).
+1. Import the `node-http-server-assignment.postman_collection.json` file that is located in the `node-http-exercise` folder.
+1. When your server is running, simply choose the route you want to test in the left pane and click on `SEND`. You should then see the response and the status code.
+
+### Extensions
```

Comment:
If you'd like to add any others, an interesting extension that will help relate this to Express later could be to serve a static file from disk. I'm sure that would be easy to lookup for the apprentice.

e.g. `/postman.json` could read and respond with the collection file.

Reply:
As mentioned in the issue comments, I'd suggest adding an extension to try some of the endpoints in a browser, especially the redirecting github one.

Reply:
Added some extensions fixes. Didn't work on the relation to Express nor serving the static file from disk, though. Let me know if you want this added.

changes done in [799330c](https://github.com/Techtonica/curriculum/pull/1406/commits/799330cf2b82a0e0d74d3cd2e258f97adc3e70db)